src/content/en/fundamentals/media/manipulating/live-effects.md

```diff
@@ -0,0 +1,376 @@
+
+For example, a brightness control can be implemented by simply taking the red, green and blue values
+of the pixel and multiplying them by a brightness value. A brightness of 0 will make the image
+entirely black. A value of 1 will leave the image unchanged.
+
+For 2D canvas:
+
+    const brightness = 1.1; // Make the image 10% brighter
+    for (let i = 0; i < imageData.data.length; i += 4) {
+      imageData.data[i] = imageData.data[i] * brightness;
+      imageData.data[i + 1] = imageData.data[i + 1] * brightness;
+      imageData.data[i + 2] = imageData.data[i + 2] * brightness;
+    }
+
+Note that the loop moves 4 bytes along at a time, but only changes three values - this is because it
+skips changing the alpha value. Also remember that a Uint8ClampedArray will round all values to
+integers, and clamp any value greater than 255.
+
+Some of these filters do take some extra information, such as the average luminance of the whole
+image, but these are things that can be calculated once for the whole image.
+
+One way of changing contrast, for example, can be to move each pixel towards  or away from some
+'gray' value, for lower or higher contrast respectively. The gray value is commonly chosen to be a
+gray color whose luminance is the median luminance of all pixels in the image.
+
+You can calculate this value once when the image is loaded and then use it each time you need to
+adjust the image effect.
+
+### Multi-pixel
+Some effects use the color of neighboring pixels when deciding the color of the current pixel.
+
+This slightly changes how you do things in the 2D canvas case because you want to be able to read
+the *original* colors of the image, and the previous example was updating the pixels in place.
+
+This is easy enough, though. When you initially create your imageData object you can make a copy of
+the data.
+
+    const originalPixels = new Uint8Array(imageData.data);
+
+Note that in the WebGL case it isn't actually possible to query the **new** color of another pixel
+from inside the shader, so you always get the original.
+
+The most common category of multi-pixel effect is called a convolution filter. You take each nearby
+pixel, including the current one, and add some multiple of the color, called the weight, to a total.
+Then you divide the result by the sum of the weights.
+
+The weights can be represented by a matrix, called a kernel, with the central value corresponding to
+the current pixel. For example, this is the kernel for a 3x3 Gaussian blur.
+
+    | 0  1  0 |
+    | 1  4  1 |
+    | 0  1  0 |
+
+So to calculate the output pixel you take the colors of the pixels above, below, to the left and to
+the right and sum them all together with four times the value of the current pixel. Then you divide
+the result by 8. You can see how the result will be a pixel that is mostly the original, but with
+the nearby pixels bleeding in.
```

Comment:
So the only pixel you're changing is the center? Do I have that right? You should show the result somehow.